### DISCUSSION TITLE
Tackling BER in Ray Tracing

### DISCUSSION THREAD

**Discussion by anindha-dhar** [NVlabs/sionna#352] [Q&A]
I have asked something before regarding BER [https://github.com/NVlabs/sionna/discussions/333](url)

But my concern is, in the ray tracing notebook, if I normalize, then a closer transmitter gives the same BER as a distant transmitter. But that's not the practical case. So we have to consider pathloss. 

**I believe the way Sionna BER designed does not necessarily represent the actual calculation of BER. Because we can not simply assume that in a practical scenario BS has a capability to increase power in an arbitrary manner to compensate for the path loss.**. 

**What I think Sionna originally considered by setting "True" to Normalize the Path loss is actually known as channel inversion power control. It assumes that BS can always change its power in such a way as such it can maintain a constant SINR.**

I might be wrong. I'm not sure about this. So I'm looking for guidance in this regard. My work is to build a digital twin, which mimics  a practical cellular network, so I'm considering the practical case. 

Any advice or suggestion will surely help me a lot. Thanks in advance.

**Comment by SebastianCa** [comment]
Hi,
As highlighted in issue #333, Sionna focuses on link-level simulations. This approach assumes the presence of a *gain control* mechanism which adjusts/equalizes the received power levels.

When you set `normalize=False`, it means that the simulation does not apply any form of gain control. As a result, only the path loss is considered, which leads to the observed poor BER. 
Enabling it with `normalize=True` simulates an ideal gain control that effectively neutralizes the path loss, hence the performance noted.

For simulations that aim to mirror real-world conditions more closely (i.e., system-level simulations), incorporating a power control strategy would be essential.

**Reply by anindha-dhar** [reply]
Thanks for you reply. It has cleared many confusions. 

Can you please suggest where can I incorporate the power control strategy? Specifically in which portion of the code should be modified to incorporate the power control strategy?

I have also adjusted the noise power `no` & `SNR ` in this code:

```
ebno_db = 2. # SNR in dB

no = ebnodb2no(ebno_db,
               pusch_transmitter._num_bits_per_symbol,
               pusch_transmitter._target_coderate,
               pusch_transmitter.resource_grid)
```

I tried with different types of values of it. But still getting a large BER.

Your guidance will surely help me a lot. Thanks in advance.
